postfix_mta_sts_resolver/responder.py

```diff
@@ -44,12 +45,24 @@ def __init__(self, cfg, loop):
                                                      timeout=zone["timeout"])))
                            for k, zone in cfg["zones"].items())
 
-        # Construct cache
-        self._cache = create_cache(cfg["cache"]["type"],
-                                   cfg["cache"]["options"])
+        self._cache = cache
         self._children = set()
         self._server = None
 
+    # Check if cached record is nonexistent or stale
+    def is_stale(self, cached):
+        if cached is None:
+            return True
+
+        ts = time.time()  # pylint: disable=invalid-name
+        if ts - cached.ts > self._grace:
+            if not self._proactive_fetch_enabled:
```

Comment:
There are few reasons why we shouldn't make exceptions for enabled proactive fetch:
1. Proactive fetch by itself does not eliminate necessity to check updates for STS policy in timely fashion.
2. It's not guaranteed policy was successfully fetched during last pre-fetch run.
3. User may effectively enable this behavior on his own setting grace time equal to policy prefetch interval. If policy was prefetched, ts will be up to date and cached record will be considered not stale.

Reply:
Good point(s). I can see a user being confused by the interaction between grace time and prefetch interval in this case however. Even more so with setting a reasonable ratio between the two values, which can be quite different depending on the use case (high vs low volume sender for example).

I guess removing this nested `if` and adding a sentence or two to the doc should be best.